src/main/groovy/com/squareup/wire/gradle/WireGeneratorTask.groovy

```diff
@@ -47,8 +46,11 @@ class WireGeneratorTask extends DefaultTask {
       for (Map.Entry<String, Collection<String>> entry : files.asMap().entrySet()) {
         String protoDir = entry.key
         List<String> args = Lists.newArrayList()
-        args.add("--proto_path=" +
-            new File(project.projectDir, protoDir).getAbsolutePath())
+        args.add("--proto_path=" + project.file(protoDir).getAbsolutePath())
```

Comment:
Using project.file instead of File() enables correct handling of absolute or relative path names.